### PR TITLE
Consistent number formatting for stacks

### PR DESCRIPTION
**Pull request by arlyon**
Adds a util class with three formatter functions:

 - runescape style (which matches the one in game)
 - "pretty" style with locale-aware formatting and 3 sig figs
 - string to number for converting the other direction (open to criticism, it was a little tricky to find a good method)

Open to suggestions for a better name. ItemManager called it "stack size" so that's what I went with.

Additionally, it updates plugins using gp to use the new formatter.

Before

image

image

After

image

image

I wish I had a better variety of stuff to show it off on but I'm a poor noob

Addresses issue #764

### CODE REVIEW

**Comment by arlyon** [comment]
I'll need to change some of the test cases since they won't pass depending on what locale they are run in

**Comment by deathbeam** [comment]
Hmm, probably examine should not be changed, you usually want to see pretty accurate price there.

**Comment by arlyon** [comment]
Maybe examine and the price check command can keep the whole price but ground items and grand exchange offers could show the abbreviated version.

**Comment by deathbeam** [comment]
I agree with ground items, but grand exchange offers probably should be accurate as well.

File: runelite-client/src/main/java/net/runelite/client/util/StackFormatter.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
+package net.runelite.client.util;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/util/StackFormatter.java]: Add the license header to each new file

File: runelite-client/src/test/java/net/runelite/client/util/StackFormatterTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			assertEquals(33_560_000, StackFormatter.stackSizeToQuantity("33.56M"));
+			assertEquals(2_000_000_000, StackFormatter.stackSizeToQuantity("2B"));
+
+			assertEquals(0, StackFormatter.stackSizeToQuantity("-0"));
+			assertEquals(-400, StackFormatter.stackSizeToQuantity("-400"));
+			assertEquals(-400_000, StackFormatter.stackSizeToQuantity("-400k"));
+			assertEquals(-40_543_000, StackFormatter.stackSizeToQuantity("-40.543m"));
+		} catch (ParseException e)
+		{
+			fail(e.getMessage());
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/util/StackFormatterTest.java]: Just make the test throw whatever, junit errors when exceptions are thrown

**Comment by arlyon** [comment]
as of now: 

stack formatter for
- ground items
- blast furnace coffer

locale formatter for
- grand exchange
- examine
- price command

**Pull request merged**